util/oidc/oidc.go

```diff
@@ -199,6 +247,11 @@ func (a *ClientApp) HandleLogin(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	returnURL := r.FormValue("return_url")
+	// Check if return_url is valid, otherwise abort processing (see #2707)
+	if !isValidRedirectURL(returnURL, []string{a.settings.URL}) {
+		http.Error(w, "Invalid return_url", http.StatusBadRequest)
```

Comment:
I was unsure what HTTP response code to chose. I picked HTTP 400 Bad Request, because that's what we basically considered it to be